### ISSUE TITLE
Different compilation results from tsc

### ISSUE THREAD

**Issue by akopchinskiy** [TypeStrong/ts-node#1093]
### Expected Behavior

no errors

### Actual Behavior

`src/api/main.ts:12:18 - error TS2339: Property 'sendFormatted' does not exist on type 'Response'.`

### Steps to reproduce the problem

try to augment 'express-serve-static-core' module through a d.ts file

### Code

```ts
// types.d.ts

declare module 'express-serve-static-core' {
    import { ResponseFormatterData } from './middlewares/response';

    interface Response {
        sendFormatted(data?: ResponseFormatterData) : Response<any>
    }
}
```

### Details

node : v14.7.0
npm : 6.14.7
tsc : 3.9.7
ts-node : v8.10.2

Win 10

### Related Issues
#391

**Comment by cspotcode** [comment]
Check this out and let us know if it solves the problem: https://github.com/TypeStrong/ts-node#help-my-types-are-missing

**Comment by cspotcode** [comment]
I'm going to assume that the link above solved the problem for you.  If it turns out this is a ts-node bug, let me know and I'll re-open this issue.

**Comment by akopchinskiy** [comment]
> I'm going to assume that the link above solved the problem for you. If it turns out this is a ts-node bug, let me know and I'll re-open this issue.

No, It doesn't. It's not only doesn't resolved the problem, but made it worse.

**Comment by cspotcode** [comment]
Reopened.  I've left it marked "needs more info" because we will need a minimal, reproducible example in order to identify and fix any bugs.
https://stackoverflow.com/help/minimal-reproducible-example
Thanks in advance.

**Comment by akopchinskiy** [comment]
@cspotcode I have managed to fix it by using `ts-mocha` package, instead of  `mocha -r ts-node/register`. But don't understand this problem. Why ts-node ignores types of some packages?
To be specific, **ts-node** ignores this package's types ("got"): https://www.npmjs.com/package/got
Actually **ts-node** can run related code without issues, but only if I'm using flag '--files'. However usage of this flag is impossible with testing frameworks, like **mocha** or **jest**.
In summary, I can say that **ts-node** default behavior  which differs from **tsc (typescript)** default behavior - is a fundamental issue, that needs to be fixed.

**Comment by cspotcode** [comment]
You can add all ts-node flags in your tsconfig file.  We also support various environment variables.  Will that enable you to use `--files` with mocha?

Our readme explains why `ts-node`'s files behavior exists, and the `files` option (which can be placed within your tsconfig file) allows ts-node to behave identically to `tsc`.  I think that covers all the bases, but let us know if we're missing anything.

https://github.com/TypeStrong/ts-node#help-my-types-are-missing

```
// in your tsconfig file
{
  "ts-node": {
    "files": true
  },
  "compilerOptions: {
    // more tsconfig stuff here
  }
}
```

You can also add `/// <reference>` directives to trigger inclusion of types.

**Comment by cspotcode** [comment]
Looks like ts-mocha uses ts-node internally, so if they are doing something with ts-node options to make stuff work, then you may be able to learn which tricks are necessary from their code: https://github.com/piotrwitek/ts-mocha/blob/master/src/index.js

**Comment by akopchinskiy** [comment]
> // in your tsconfig file
> {
>   "ts-node": {
>     "files": true
>   },
>   "compilerOptions: {
>     // more tsconfig stuff here
>   }
> }

Just did that.

`tsc --project tsconfig.dev.json`
Gives no errors.

`ts-node --project tsconfig.dev.json ./src/api/main.ts`
Gives:
```cmd
// error TS2339: Property 'body' does not exist on type 'CancelableRequest<Response<unknown>>'.

const { body } = await got.get(this.url, {
```

My tsconfig.json and tconfig.dev.json:

```json
// tsconfig.json

{
  "compilerOptions": {
    /* Basic Options */
    // "incremental": true,                   /* Enable incremental compilation */
    "target": "ES2020",
    /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019' or 'ESNEXT'. */
    "module": "CommonJS",
    /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
    "lib": [
      "ES5",
      "ES6",
      "ES7",
      "ES2018",
      "ES2019",
      "ES2020"
    ],
    "sourceMap": true,
    /* Generates corresponding '.map' file. */
    // "outFile": "./",                       /* Concatenate and emit output to single file. */
    "outDir": "./dist",
    /* Redirect output structure to the directory. */
    "rootDir": ".",
    /* Specify the root directory of input files. Use to control the output directory structure with --outDir. */
    // "composite": true,                     /* Enable project compilation */
    // "tsBuildInfoFile": "./",               /* Specify file to store incremental compilation information */
    "removeComments": true,
    /* Do not emit comments to output. */
    "noEmit": true,
    /* Do not emit outputs. */
    // "importHelpers": true,                 /* Import emit helpers from 'tslib'. */
    // "downlevelIteration": true,            /* Provide full support for iterables in 'for-of', spread, and destructuring when targeting 'ES5' or 'ES3'. */
    // "isolatedModules": true,               /* Transpile each file as a separate module (similar to 'ts.transpileModule'). */
    /* Strict Type-Checking Options */
//    "strict": true,    /* Enable all strict type-checking options. */
    "strictBindCallApply": true,
//    "strictFunctionTypes": true,
    "alwaysStrict": true,    /* Parse in strict mode and emit "use strict" for each source file. */
    "noImplicitAny": false,
    /* Raise error on expressions and declarations with an implied 'any' type. */
    // "strictFunctionTypes": true,           /* Enable strict checking of function types. */
    // "strictBindCallApply": true,           /* Enable strict 'bind', 'call', and 'apply' methods on functions. */
    // "strictPropertyInitialization": true,  /* Enable strict checking of property initialization in classes. */
    "noImplicitThis": true,
    /* Raise error on 'this' expressions with an implied 'any' type. */
    /* Additional Checks */
    //    "noUnusedLocals": true,
    /* Report errors on unused locals. */
    //    "noUnusedParameters": true,
    /* Report errors on unused parameters. */

    // "noImplicitReturns": true,             /* Report error when not all code paths in function return a value. */
    // "noFallthroughCasesInSwitch": true,    /* Report errors for fallthrough cases in switch statement. */
    "forceConsistentCasingInFileNames": true,
    /* Module Resolution Options */
    "moduleResolution": "node",
    /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
    "baseUrl": "./src",
    /* Base directory to resolve non-absolute module names. */
    /* List of folders to include type definitions from. */
    // "types": [],                           /* Type declaration files to be included in compilation. */
    "allowSyntheticDefaultImports": true,
    /* Allow default imports from modules with no default export. This does not affect code emit, just typechecking. */
    "esModuleInterop": true,
    /* Enables emit interoperability between CommonJS and ES Modules via creation of namespace objects for all imports. Implies 'allowSyntheticDefaultImports'. */
    // "preserveSymlinks": true,              /* Do not resolve the real path of symlinks. */
    // "allowUmdGlobalAccess": true,          /* Allow accessing UMD globals from modules. */
    /* Source Map Options */
    // "sourceRoot": "",                      /* Specify the location where debugger should locate TypeScript files instead of source locations. */
    // "mapRoot": "",                         /* Specify the location where debugger should locate map files instead of generated locations. */
    // "inlineSourceMap": true,               /* Emit a single file with source maps instead of having a separate file. */
    // "inlineSources": true,                 /* Emit the source alongside the sourcemaps within a single file; requires '--inlineSourceMap' or '--sourceMap' to be set. */
    /* Experimental Options */
    "experimentalDecorators": true,
    /* Enables experimental support for ES7 decorators. */
    "emitDecoratorMetadata": true,
    /* Enables experimental support for emitting type metadata for decorators. */
    "downlevelIteration": true,
    "skipLibCheck": true,
    "watch": false
  },
  "ts-node": {
    "files": true
  },
  "include": [
    "src/**/*"
  ],
  "exclude": [
    "node_modules"
  ]
}

// tsconfig.dev.json

{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "noEmit": true,
    "watch": true
  }
}

```

**Comment by akopchinskiy** [comment]
@cspotcode 
I have created test repo locally with the same versions of ts, ts-node and got packages, also with the same `ts-config.json` . No errors. 
So it's not the ts-node issue, something probably with my IDE or npm cache. 
I tried to install it with npm, then removed node_modules and reinstalled with pnpm and get this error.
So the problem is in pnpm.

**Comment by cspotcode** [comment]
Do you get the error with `ts-node`, `tsc`, or both?

You mention a test repo; is that pushed anywhere so that someone else can clone, see the same error, and inspect?
When we ask for a minimal, reproducible example, the goal is for us to see exactly the same error and figure out the root cause.  It also makes it easier for us to propose a fix, if appropriate, because we can link to specific lines or even push a fixed branch.
https://stackoverflow.com/help/minimal-reproducible-example

**Comment by akopchinskiy** [comment]
@cspotcode Problem was only with ts-node. tsc works like a charm, but at the end (when I created a separated repo) ts-node was correct about that type error and tsc don't.
> a minimal, reproducible example
The problem is - **I CAN NOT REPRODUCE IT INTENTIONALLY**.

**Comment by cspotcode** [comment]
Excellent, with the reproduction I can see the error on my end, and I can verify that this has been fixed in #970, which is currently in review.  Once it's reviewed, we can merge it and publish v9.0.0 which should fix this for you.

I verified by downloading the `ts-node` package built by CI for that branch (`ts-node-packed.tgz` from https://github.com/TypeStrong/ts-node/actions/runs/204543752) and installing it via pnpm in your reproduction.
